### DISCUSSION TITLE
Does Longhorn has a system to automatically balance volumes between disks.

### DISCUSSION THREAD

**Discussion by iosifnicolae2** [longhorn/longhorn#3887] [Q&A]
Hey,

Does Longhorn has a mechanism to automatically balance volumes between disks/nodes?

For example, we want to move some volumes from `dedicated-worker-1-fast-disk-1` to `dedicated-worker-1-fast-disk-2`
![image](https://user-images.githubusercontent.com/7108834/164718774-042572b3-0ebb-4028-816c-28cdee692a06.png)

Thank you!

**Comment by iosifnicolae2** [comment]
Also, it would be very useful to have the ability to evict a volume from the current node and move it to another node.

**Reply by voarsh2** [reply]
Evicting from a disk is a working feature.

**Comment by joshimoo** [comment]
Perhaps the auto balance replica feature works for you, If I remember right it only considers nodes and not disks.
https://longhorn.io/docs/1.2.4/high-availability/auto-balance-replicas/

If after testing you find it lacking or require disks balancing options please open a github issue as an enhancement request, and we can consider it during the replica scheduling refactor #3840

**Reply by iosifnicolae2** [reply]
I've created https://github.com/longhorn/longhorn/issues/4105.

Thank you!
